ccm-connector/src/test/java/com/sequenceiq/cloudbreak/ccmimpl/ccmv2/CcmV2RetryingClientTest.java

```diff
@@ -0,0 +1,79 @@
+package com.sequenceiq.cloudbreak.ccmimpl.ccmv2;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.sequenceiq.cloudbreak.ccm.exception.CcmV2Exception;
+import com.sequenceiq.cloudbreak.ccmimpl.ccmv2.config.GrpcCcmV2Config;
+
+@ExtendWith(MockitoExtension.class)
+class CcmV2RetryingClientTest {
+
+    private static final String TEST_REQUEST_ID = "requestId";
+
+    private static final String TEST_ACCOUNT_ID = "us-west-1:e7b1345f-4ae1-4594-9113-fc91f22ef8bd";
+
+    private static final String TEST_AGENT_CRN = "testAgentCrn";
+
+    private static final String TEST_ENVIRONMENT_CRN = "environmentCrn";
+
+    private static final String DOMAIN = "domain";
+
+    private static final String KEY_ID = "key";
+
+    private static final String HMAC = "hmac";
+
+    @Mock
+    private CcmV2Client ccmV2Client;
+
+    @Mock
+    private GrpcCcmV2Config grpcCcmV2Config;
+
+    @InjectMocks
+    private CcmV2RetryingClient underTest;
```

Comment:
Wouldn't we have liked to mock the `RetryTemplate` in the client to avoid spring related runtime issues and the flakiness of the test?